core/src/sumeragi/main_loop.rs

```diff
@@ -420,6 +431,50 @@ fn commit_block<F: FaultInjection>(
     cache_transaction(state, sumeragi);
 }
 
+fn replace_top_block<F: FaultInjection>(
+    sumeragi: &SumeragiWithFault<F>,
+    state: &mut State,
+    block: impl Into<VersionedCommittedBlock>,
+) {
+    let committed_block = block.into();
+    let block_hash = committed_block.hash();
+
+    state.wsv = state.finalized_wsv.clone();
+    state
+        .wsv
+        .apply(&committed_block)
+        .expect("Failed to apply block on WSV. This is absolutely not acceptable.");
```

Comment:
```suggestion
        .expect("Failed to apply block to WSV. Bailing.");
```

AFAIK, this is common terminology for "It's too dangerous to continue running, so we're stopping the execution".